torchvision/transforms/transforms.py

```diff
@@ -527,6 +527,68 @@ def __repr__(self):
         return self.__class__.__name__ + '(p={})'.format(self.p)
 
 
+class RandomPerspective(object):
+    """Performs Perspective transformation of the given PIL Image randomly with a given probability.
+
+    Args:
+        p (float): probability of the image being perspectively transformed. Default value is 0.5
+        interpolation : Default- Image.BICUBIC
+    """
+
+    def __init__(self, interpolation=Image.BICUBIC, p=0.5):
+        self.p = p
+        self.interpolation = interpolation
+
+    def __call__(self, img):
+        """
+        Args:
+            img (PIL Image): Image to be Perspectively transformed.
+
+        Returns:
+            PIL Image: Random perspectivley transformed image.
+        """
+        if not F._is_pil_image(img):
+            raise TypeError('img should be PIL Image. Got {}'.format(type(img)))
+
+        if random.random() < self.p:
+            width, height = img.size
+            coeffs = self.get_params(width, height)
+            return F.perspective(img, coeffs, self.interpolation)
+        return img
+
+    @staticmethod
+    def get_params(width, height):
+        """Get parameters for ``perspective`` for a random perspective transform.
+
+        Args:
+            width : width of the image.
+            height : height of the image.
+        Returns:
+            tuple: 8-tuple (a, b, c, d, e, f, g, h) which contains the coefficients
+                    for a perspective transform.
+        """
+
+        topleft = (random.randint(0, int(width / 2)), random.randint(0, int(height / 2)))
```

Comment:
Minor optimization: compute width / 2 and height / 2 in separate variables since they're reused.

Something like:
```python
half_width, half_height = width // 2, height // 2
topleft = (random.randint(0, half_width), random.randint(0, half_height))
```